runtime/src/cost_model.rs

```diff
@@ -180,9 +180,19 @@ impl CostModel {
             feature_set.is_active(&add_set_tx_loaded_accounts_data_size_instruction::id()),
         );
 
-        // if tx contained user-space instructions and a more accurate estimate available correct it
-        if bpf_costs > 0 && result.is_ok() {
-            bpf_costs = budget.compute_unit_limit
+        // if failed to process compute_budget instructions, the transaction will not be executed
+        // by `bank`, therefore it should be considered as no execution cost by cost model.
+        match result {
+            Ok(_) => {
+                // if tx contained user-space instructions and a more accurate estimate available correct it
+                if bpf_costs > 0 {
+                    bpf_costs = budget.compute_unit_limit
+                }
+            }
+            Err(_) => {
```

Comment:
what about `data_bytes_cost`?

Reply:
my thinking is `data_bytes_cost` is to model cost of serializing instruction data, which happens regardless if instruction fails or succeeds, so it's cost remain the same.

Reply:
So right now, these costs will not (permanently) go into the cost tracking, since they aren't executed, and thus not committed, we remove the costs from the tracker after processing the batch. After the feature activation for not updating qos, these costs will stay there. Argument for data_bytes_cost is that we are doing some ops regardless due to instruction data size.

I can buy that. It seems to me a similar argument could be made for the builtin costs of the compute budget instructions though, so those are processed until first error.

Reply:
yea, I see your argument. The compute budget is tricky because it is processed multiple times for a tx (once here for block packing, once at bank for execution, once at accounts for payer fee checking, maybe other places). I guess those occurrences are considered as part of basic processes as tx moves through pipeline. 

The execution_cost (the combine of `builtin_execution_cost` and `bpf_execution_cost`) are to capture CUs `runtime` consumes, directly maps to requested `cu_limit`. 

The logic there is if a TX is not going to hit `runtime`, there will be no execution_cost.